upgrade/v121to122/upgrade.go

```diff
@@ -0,0 +1,140 @@
+package v121to122
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/longhorn/longhorn-manager/engineapi"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+const (
+	upgradeLogPrefix = "upgrade from v1.2.1 to v1.2.2: "
+)
+
+func UpgradeCRs(namespace string, lhClient *lhclientset.Clientset) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, upgradeLogPrefix+"UpgradeCRs failed")
+	}()
+	if err := upgradeBackups(namespace, lhClient); err != nil {
+		return err
+	}
+	if err := upgradeEngines(namespace, lhClient); err != nil {
+		return err
+	}
+	return nil
+}
+
+func upgradeBackups(namespace string, lhClient *lhclientset.Clientset) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "upgrade backups failed")
+	}()
+
+	// Copy backupStatus from engine CRs to backup CRs
+	backups, err := lhClient.LonghornV1beta1().Backups(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Loop all the backup CRs
+	for _, backup := range backups.Items {
+		// Get volume name from label
+		volumeName, exist := backup.Labels[types.LonghornLabelBackupVolume]
+		if !exist {
+			continue
+		}
+
+		// Get the corresponding volume's engine
+		engines, err := lhClient.LonghornV1beta1().Engines(namespace).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: types.LonghornLabelVolume + "=" + volumeName,
+		})
+		if err != nil {
+			continue
```

Comment:
Suggest don't silently swallow any error message even you don't want it block the upgrade path. It's better to have a log for it.

Reply:
Okay, I'll add check if the `err` is `apierrors.IsNotFound`.
If yes, it means the Backup CR doesn't have corresponding Volume/Engine within the cluster, we should continue the next Backup CR.
If no, it means something error when List the engines, will return an error.

Reply:
Change to directly error out.